codec-http2/src/main/java/io/netty/handler/codec/http2/ReadOnlyHttp2Headers.java

```diff
@@ -776,6 +781,55 @@ public String toString() {
         return builder.append(']').toString();
     }
 
+    private final class ReadOnlyValueIterator implements Iterator<CharSequence> {
+        private int i;
+        private final int nameHash;
+        private final CharSequence name;
+        private AsciiString[] current = pseudoHeaders.length != 0 ? pseudoHeaders : otherHeaders;
+        private AsciiString next;
+
+        ReadOnlyValueIterator(CharSequence name) {
+            nameHash = AsciiString.hashCode(name);
+            this.name = name;
+            calculateNext();
+        }
+
+        @Override
+        public boolean hasNext() {
+            return next != null;
+        }
+
+        @Override
+        public CharSequence next() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "NoSuchElementException" for iteration beyond the end of the collection. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2272)

Reply:
@Scottmitch  please do as netkins said :)

Reply:
as I said I considered doing this before in https://github.com/netty/netty/pull/7216#issuecomment-329922454 ... I've always had a thing against the duplication in responsibility on the Iterator interface ... this condition has to be verified by users of the interface and by the implementations of the interface. However I guess this can be avoided in the future by overriding `forEachRemaining`.